internal/os/smb/api.go

```diff
@@ -38,6 +38,13 @@ func (APIImplementor) IsSmbMapped(remotePath string) (bool, error) {
 // alpha to merge the paths.
 // TODO (for beta release): Merge the link paths - os.Symlink and Powershell link path.
 func (APIImplementor) NewSmbLink(remotePath, localPath string) error {
+
+	if !strings.HasSuffix(remotePath, "\\") {
```

Comment:
Do we expect this code path to ever be used for a file symlink? If so, adding a slash to the end would break it.

Reply:
We technically only need the slash added if the path is to the share root, also (e.g. `\\server\share` doesn't work currently, but `\\server\share\dir` does).

Reply:
On the node I was testing these changes out on only the symlink created against the root share used this codepath.

Here is the state of some symlinks created on that node

```
azureuser@1876k8s00000000 c:\k>dir "c:\var\lib\kubelet\pods\7819fd8b-cdbd-4292-8554-769ad4af543a\volumes\kubernetes.io~csi\pvc-ee3f6a92-f195-4d9e-841b-6966e6839829"
 Volume in drive C is Windows
 Volume Serial Number is CA78-414E

 Directory of c:\var\lib\kubelet\pods\7819fd8b-cdbd-4292-8554-769ad4af543a\volumes\kubernetes.io~csi\pvc-ee3f6a92-f195-4d9e-841b-6966e6839829

10/19/2020  11:11 PM    <DIR>          .
10/19/2020  11:11 PM    <DIR>          ..
10/19/2020  11:11 PM    <SYMLINKD>     mount [c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-ee3f6a92-f195-4d9e-841b-6966e6839829\globalmount]
10/19/2020  11:10 PM               340 vol_data.json
               1 File(s)            340 bytes
               3 Dir(s)  117,465,772,032 bytes free

azureuser@1876k8s00000000 c:\k>dir c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-ee3f6a92-f195-4d9e-841b-6966e6839829
 Volume in drive C is Windows
 Volume Serial Number is CA78-414E

 Directory of c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-ee3f6a92-f195-4d9e-841b-6966e6839829

10/19/2020  11:11 PM    <DIR>          .
10/19/2020  11:11 PM    <DIR>          ..
10/19/2020  11:11 PM    <SYMLINKD>     globalmount [\\faf486191df3c4f0fad548f.file.core.windows.net\pvc-ee3f6a92-f195-4d9e-841b-6966e6839829\]
10/19/2020  11:10 PM               135 vol_data.json
               1 File(s)            135 bytes
               3 Dir(s)  117,465,772,032 bytes free
```

Reply:
I believe this codepath will only be used for creating links to SMB shares.
Will try and confirm.

Reply:
@andyzhangx 

Also want to verify there should be no problem adding trailing backslash in any case. Is there any possibility that remote path is a file?

Reply:
smb remote path is a folder, this PR lgtm, thanks